### PR TITLE
Feature profile details integration #35

### PR DESCRIPTION
**Pull request by seblaifa3315**
### What this PR does (required):
- Made profile details page dynamic based on user using params which should be pushed from listing profile page
-  Added functionality to create a request to the specific sitter based on the date picker

### Any information needed to test this feature (required):
- login with email "johndoe@gmail.com" and password "johndoe"
- go to the path="/profile-details/:sitterId"

**Here are some sitterId to use as params:**
62042faa5acc1804d4ad8874
620545804d573a441a7ac55f
6204307506cc5874cc5022ad
620544e34d573a441a7ac547
6205450b4d573a441a7ac54d
6205452d4d573a441a7ac553
620545594d573a441a7ac559

### CODE REVIEW

File: server/controllers/request 2.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+const Request = require("../models/Request");
+
+exports.createRequest = async (req, res) => {
```
Comment (ezeilo-su) [inline server/controllers/request 2.js:5]: Let's wrap this handler with the `asyncHandler` middleware.

File: server/controllers/request 2.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        const request = await Request.create(req.body);
+
+        res.status(200).json(request);
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+};
+
+exports.getRequests = async (req, res) => {
```
Comment (ezeilo-su) [inline server/controllers/request 2.js:22]: The same here as with the comment above

File: server/controllers/request 2.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            );
+            return res.status(200).json(requestsWithSitterInfo);
+        }
+            
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+};
+
+exports.updateRequest = async (req, res) => {
```
Comment (ezeilo-su) [inline server/controllers/request 2.js:68]: This one too

File: server/controllers/request.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        if(theRequest.sitterId.valueOf() !== req.user.id) return res.status(404).json({ message: "This sitter doesn't own this request" });
+
+        theRequest.status = newStatus;
+        await theRequest.save();       
+
+        res.status(200).json(theRequest);
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+};
```
Comment (ezeilo-su) [inline server/controllers/request.js:87]: It seems your duplicated these ones

File: server/models/Request 3.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        timestamps: true,
+    }
+);
+
+function dateValidator(value) {
+    return this.start <= value;
+}
+
+module.exports = Request = mongoose.model("Request", requestSchema);
```
Comment (ezeilo-su) [inline server/models/Request 3.js:57]: Please remove duplicate schema

File: server/routes/request 2.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { createRequest, getRequests, updateRequest } = require("../controllers/request");
+
+router.route("/").post(protect, createRequest);
+router.route("/").get(protect, getRequests);
+router.route("/:requestId").put(protect, updateRequest);
+
+module.exports = router;
```
Comment (ezeilo-su) [inline server/routes/request 2.js:10]: Duplicate file, please remove.

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Hey @0916dhkim, please remove all the duplicate files in your project.

File: client/src/App.tsx [hunk 7]
```diff
@@ -28,6 +29,7 @@ function App(): JSX.Element {
                 <Route exact path="/signup" component={Signup} />
                 <Route exact path="/dashboard" component={Dashboard} />
                 <Route path="/profile/settings" component={Settings} />
+                <Route path="/profile-details/:sitterId" component={ProfileDetails} />
```
Comment (rajivtitus) [inline client/src/App.tsx:32]: 👍🏽

File: server/controllers/request.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+const Request = require("../models/Request");
+
+exports.createRequest = async (req, res) => {
+    const { ownerId, sitterId, start, end, animalType, usefulInfo } = req.body;
+    console.log(req.body);
```
Comment (rajivtitus) [inline server/controllers/request.js:7]: Please remove logs to the console

**Review by rajivtitus — COMMENTED** [review]
Good work with this! I noticed that we have duplicates for the request model, routes and controllers - please delete the outdated versions and update this PR 👍🏽

File: client/src/components/Navbar/Navbar.tsx [hunk 9]
```diff
@@ -126,7 +125,11 @@ const Navbar: React.FC = () => {
     handleClose();
     logout();
   };
-  const [profile, setProfile] = useState<Profile>({ accountType: 'initial' });
+
+  interface IAccountType {
```
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx:129]:
```suggestion
  interface AccountType {
```
Please avoid using 'I' as a prefix for interfaces in JS

File: server/models/Profile.js [hunk 10]
```diff
@@ -17,6 +17,7 @@ const profileSchema = new mongoose.Schema({
   gender: {
     type: String,
     enum: ['male', 'female', 'other'],
+    default: 'other',
```
Comment (rajivtitus) [inline server/models/Profile.js:20]: We don't want to have a default gender 😄

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making those changes 👍🏽

**Pull request merged**